File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/personal/PersonalListAccountsTest.java
package opensdk.sdk.apis.personal;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.PersonalListAccountsResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Personal RPC Test")
public class PersonalListAccountsTest {

    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC personal_listAccounts")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        PersonalListAccountsResponse response = sdk.personal.listAccounts()
                .send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/personal/PersonalListAccounts.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.PersonalApi(new OpenSdk.ApiClient(RPC));

describe('personal_listAccounts API', () => {
    test('should return personal_listAccounts', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(Array.isArray(data)).toBe(true);
        
            done();
        };

        sdk.listAccounts({}, callbackOne);
    });
});

